cineast-core/src/main/java/org/vitrivr/cineast/core/util/TextDetector_EAST.java

```diff
@@ -0,0 +1,163 @@
+     */
+    public TextDetector_EAST initialize() {
+        return initialize("resources/TextSpotter/frozen_east_text_detection.pb");
+    }
+
+    private Pair<List<Float>, List<RotatedRect>> decode(Mat scores, Mat geometry) {
+        List<Float> confidences = new ArrayList<>();
+        List<RotatedRect> boxes = new ArrayList<>();
+        int W = geometry.cols();
+        int H = geometry.rows() / 5;
+
+        for (int y = 0; y < H; ++y) {
+            Mat scoresData = scores.row(y);
+            Mat x0Data = geometry.submat(0, H, 0, W).row(y);
+            Mat x1Data = geometry.submat(H, 2 * H, 0, W).row(y);
+            Mat x2Data = geometry.submat(2 * H, 3 * H, 0, W).row(y);
+            Mat x3Data = geometry.submat(3 * H, 4 * H, 0, W).row(y);
+            Mat anglesData = geometry.submat(4 * H, 5 * H, 0, W).row(y);
+
+            for (int x = 0; x < W; ++x) {
+                double score = scoresData.get(0, x)[0];
+                if (score >= this.confThreshold) {
+                    double offsetX = x * 4.0;
+                    double offsetY = y * 4.0;
+                    double angle = anglesData.get(0, x)[0];
+                    double cosA = Math.cos(angle);
+                    double sinA = Math.sin(angle);
+                    double x0 = x0Data.get(0, x)[0];
+                    double x1 = x1Data.get(0, x)[0];
+                    double x2 = x2Data.get(0, x)[0];
+                    double x3 = x3Data.get(0, x)[0];
+                    double h = x0 + x2;
+                    double w = x1 + x3;
+                    Point offset = new Point(offsetX + cosA * x1 + sinA * x2, offsetY - sinA * x1 + cosA * x2);
+                    Point p1 = new Point(-1 * sinA * h + offset.x, -1 * cosA * h + offset.y);
+                    Point p3 = new Point(-1 * cosA * w + offset.x,      sinA * w + offset.y); // original trouble here !
+                    RotatedRect r = new RotatedRect(new Point(0.5 * (p1.x + p3.x), 0.5 * (p1.y + p3.y)), new Size(w, h), -1 * angle * 180 / Math.PI);
+                    boxes.add(r);
+                    confidences.add((float) score);
+                }
+            }
+        }
+        return new Pair<>(confidences, boxes);
+    }
+
+    /**
+     * detect takes an image and returns the detect text as rotated rectangles
+     * @param frame The image on which the detectin should be done
+     * @return The coordinates of the detected text instances (as rotated rectangles)
+     */
+    public Point[][] detect(Mat frame) {
+        Imgproc.cvtColor(frame, frame, Imgproc.COLOR_RGBA2RGB);
+
+        Size size = new Size(frame.width() - (frame.width() % 32), frame.height() - (frame.height() % 32));
+        int H = (int)(size.height / 4);
+        Mat blob = Dnn.blobFromImage(frame, 1.0,size, new Scalar(123.68, 116.78, 103.94), true, false);
+        this.model.setInput(blob);
+        List<Mat> outs = new ArrayList<>(2);
+
+        this.model.forward(outs, this.outNames);
```

Comment:
@silvanheller Yes, the detector and recognizer can be done in a reusable and thread-safe way. However, if I'm not mistaken this would require me to synchronize the code from line 118 to 121. Since this is an especially computationally costly piece of code, this would make the scene text extraction considerably slower.